src/subdomains/generic/user/models/user-data/user-data.service.ts

```diff
@@ -201,16 +172,22 @@ export class UserDataService {
 
     userData = await this.userDataRepo.save(Object.assign(userData, dto));
 
-    if (kycChanged) await this.kycNotificationService.kycChanged(userData, userData.kycLevel);
+    if (dto.kycLevel && dto.kycLevel !== userData.kycLevel)
+      await this.kycNotificationService.kycChanged(userData, userData.kycLevel);
```

Comment:
Is a bit hard to detect a change, if the DTO is already applied to the entity 😉

Reply:
Yeah, right. I think we can remove the check if kycLevel changed, so `if(dto.kycLevel)` should be enough, wdyt?